### PR TITLE
W2-phongle-Create recyclerview screen

### PR DESCRIPTION
**Pull request by at-phongle**
#### Related issues (Optional)
Task # 3

#### Description (Required)
- [x] Create RecyclerView Screen

### CODE REVIEW

File: app/src/at-phongle/java/vn/asiantech/internship/recyclerview/models/PostAdapter.java [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+
+import vn.asiantech.internship.R;
+
+public class PostAdapter extends RecyclerView.Adapter<PostAdapter.PostViewHolder> {
+    private List<Posts> mPostList;
+    private OnItemClickListener mOnItemClickListener;
+
+    PostAdapter(List<Posts> mPostList, OnItemClickListener mOnItemClickListener) {
+        this.mPostList = mPostList;
+        this.mOnItemClickListener = mOnItemClickListener;
```
Comment (at-huongnguyen) [inline app/src/at-phongle/java/vn/asiantech/internship/recyclerview/models/PostAdapter.java]: Remove this

File: app/src/at-phongle/java/vn/asiantech/internship/recyclerview/models/PostAdapter.java [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+            mTvLikeNumber = itemView.findViewById(R.id.tvLikeNumber);
+            mTvDescription = itemView.findViewById(R.id.tvDescription);
+        }
+
+        private void initListener() {
+            mImgLike.setOnClickListener(this);
+            mImgDislike.setOnClickListener(this);
+        }
+
+        private void onBindData(Context context) {
```
Comment (at-huongnguyen) [inline app/src/at-phongle/java/vn/asiantech/internship/recyclerview/models/PostAdapter.java]: Remove param context and use itemView.getContext()

File: app/src/at-phongle/java/vn/asiantech/internship/recyclerview/models/PostAdapter.java [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+            Posts posts = mPostList.get(getAdapterPosition());
+            mTvSubject.setText(posts.getSubject());
+            mTvLikeNumber.setText(String.valueOf(posts.getLikeNumber()));
+            mTvDescription.setText(posts.getDescription());
+            setColor(posts.getLikeNumber(), context);
+        }
+
+        private void setColor(int i, Context context) {
```
Comment (at-huongnguyen) [inline app/src/at-phongle/java/vn/asiantech/internship/recyclerview/models/PostAdapter.java]: Rename `i` to `likeNumber`

File: app/src/at-phongle/res/layout/activity_post.xml [hunk 4]
```diff
@@ -0,0 +1,10 @@
+<?xml version="1.0" encoding="utf-8"?>
+<LinearLayout xmlns:android="http://schemas.android.com/apk/res/android"
```
Comment (at-huongnguyen) [inline app/src/at-phongle/res/layout/activity_post.xml]: Remove ViewGroup

File: app/src/at-phongle/res/layout/activity_post.xml [hunk 5]
```diff
@@ -0,0 +1,10 @@
...
+    android:orientation="vertical"
+    android:layout_width="match_parent"
+    android:layout_height="match_parent">
+    <android.support.v7.widget.RecyclerView
+        android:id="@+id/rvPosts"
```
Comment (at-huongnguyen) [inline app/src/at-phongle/res/layout/activity_post.xml]: recyclerViewPost

File: app/src/at-phongle/res/layout/item_post.xml [hunk 6]
```diff
@@ -0,0 +1,61 @@
+<?xml version="1.0" encoding="utf-8"?>
+<android.support.v7.widget.CardView xmlns:android="http://schemas.android.com/apk/res/android"
+    android:layout_width="match_parent"
+    android:layout_height="wrap_content">
+
+    <RelativeLayout xmlns:android="http://schemas.android.com/apk/res/android"
```
Comment (at-huongnguyen) [inline app/src/at-phongle/res/layout/item_post.xml]: Move attr `xmlns:android="http://schemas.android.com/apk/res/android"` to ViewGroup

File: app/src/main/res/layout/activity_test.xml [hunk 7]
```diff
@@ -1,9 +1,6 @@
 <?xml version="1.0" encoding="utf-8"?>
```
Comment (at-huongnguyen) [inline app/src/main/res/layout/activity_test.xml]: Remove activity_test file

**Comment by at-phongle** [comment]
@at-huongnguyen  i fixed conflix .Please review this .Thanks

**Comment by at-phongle** [comment]
@at-huongnguyen  please review this . tks

**Comment by at-huongnguyen** [comment]
@at-phongle 
Please resolve conflicts

**Comment by at-phongle** [comment]
@at-huongnguyen  i  fixed conflix . please review this.thanks

File: app/src/at-phongle/java/vn/asiantech/internship/recyclerview/PostAdapter.java [hunk 8]
```diff
@@ -0,0 +1,108 @@
...
+import vn.asiantech.internship.R;
+import vn.asiantech.internship.models.Posts;
+
+public class PostAdapter extends RecyclerView.Adapter<PostAdapter.PostViewHolder> {
+    private List<Posts> mPostList;
+    private OnItemClickListener mOnItemClickListener;
+
+    PostAdapter(List<Posts> mPostList, OnItemClickListener mOnItemClickListener) {
+        this.mPostList = mPostList;
+        this.mOnItemClickListener = mOnItemClickListener;
```
Comment (at-huongnguyen) [inline app/src/at-phongle/java/vn/asiantech/internship/recyclerview/PostAdapter.java]: Remove `this.`

File: app/src/at-phongle/java/vn/asiantech/internship/recyclerview/PostAdapter.java [hunk 9]
```diff
@@ -0,0 +1,108 @@
...
+import java.util.List;
+
+import vn.asiantech.internship.R;
+import vn.asiantech.internship.models.Posts;
+
+public class PostAdapter extends RecyclerView.Adapter<PostAdapter.PostViewHolder> {
+    private List<Posts> mPostList;
+    private OnItemClickListener mOnItemClickListener;
+
+    PostAdapter(List<Posts> mPostList, OnItemClickListener mOnItemClickListener) {
```
Comment (at-huongnguyen) [inline app/src/at-phongle/java/vn/asiantech/internship/recyclerview/PostAdapter.java]: postList

File: app/src/at-phongle/java/vn/asiantech/internship/recyclerview/PostAdapter.java [hunk 10]
```diff
@@ -0,0 +1,108 @@
...
+
+        private void onBindData() {
+            Posts posts = mPostList.get(getAdapterPosition());
+            mTvSubject.setText(posts.getSubject());
+            mTvLikeNumber.setText(String.valueOf(posts.getLikeNumber()));
+            mTvDescription.setText(posts.getDescription());
+            setColor(posts.getLikeNumber(), itemView.getContext());
+        }
+
+        private void setColor(int likeNumber, Context context) {
```
Comment (at-huongnguyen) [inline app/src/at-phongle/java/vn/asiantech/internship/recyclerview/PostAdapter.java]: Should rename to displayLikeNumber

File: app/src/at-phongle/res/layout/activity_post.xml [hunk 11]
```diff
@@ -0,0 +1,11 @@
+<?xml version="1.0" encoding="utf-8"?>
+<LinearLayout xmlns:android="http://schemas.android.com/apk/res/android"
+    android:orientation="vertical"
+    android:layout_width="match_parent"
+    android:layout_height="match_parent"
+    android:background="@color/bg_recyclerView">
```
Comment (at-huongnguyen) [inline app/src/at-phongle/res/layout/activity_post.xml]: Remove ViewGroup

File: app/src/at-phongle/java/vn/asiantech/internship/recyclerview/PostActivity.java [hunk 12]
```diff
@@ -0,0 +1,63 @@
...
+    @Override
+    protected void onCreate(@Nullable Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_post);
+        initView();
+        initData();
+        initAdapter();
+    }
+
+    private void initView() {
```
Comment (at-huongnguyen) [inline app/src/at-phongle/java/vn/asiantech/internship/recyclerview/PostActivity.java]: initViews

**Comment by at-phongle** [comment]
@at-huongnguyen  please review this. thanks.

File: app/src/at-phongle/java/vn/asiantech/internship/models/Posts.java [hunk 13]
```diff
@@ -0,0 +1,37 @@
...
+
+    public int getLikeNumber() {
+        return likeNumber;
+    }
+
+    public void setLikeNumber(int likeNumber) {
+        this.likeNumber = likeNumber;
+    }
+
+    public Posts(String subject, String description, int likeNumber) {
```
Comment (at-vinhhuynh) [inline app/src/at-phongle/java/vn/asiantech/internship/models/Posts.java]: move this method to line 7

File: app/src/at-phongle/java/vn/asiantech/internship/models/Posts.java [hunk 14]
```diff
@@ -0,0 +1,37 @@
+package vn.asiantech.internship.models;
+
+public class Posts {
```
Comment (at-vinhhuynh) [inline app/src/at-phongle/java/vn/asiantech/internship/models/Posts.java]: change class name to `Post`

File: app/src/at-phongle/java/vn/asiantech/internship/recyclerview/PostActivity.java [hunk 15]
```diff
@@ -0,0 +1,63 @@
...
+        mRecyclerViewPost = findViewById(R.id.recyclerViewPosts);
+    }
+
+    private void initAdapter() {
+        mPostAdapter = new PostAdapter(mPostList, this);
+        mRecyclerViewPost.setLayoutManager(new LinearLayoutManager(this));
+        mRecyclerViewPost.setAdapter(mPostAdapter);
+    }
+
+    public void initData() {
```
Comment (at-vinhhuynh) [inline app/src/at-phongle/java/vn/asiantech/internship/recyclerview/PostActivity.java]: change `public` to `private`

File: app/src/main/AndroidManifest.xml [hunk 16]
```diff
@@ -13,4 +13,4 @@
         </activity>
     </application>
 
-</manifest>
\ No newline at end of file
+</manifest>
```
Comment (at-vinhhuynh) [inline app/src/main/AndroidManifest.xml]: revert this file

File: app/src/main/res/mipmap-anydpi-v26/ic_launcher_round.xml [hunk 17]
```diff
@@ -2,4 +2,4 @@
 <adaptive-icon xmlns:android="http://schemas.android.com/apk/res/android">
     <background android:drawable="@drawable/ic_launcher_background" />
     <foreground android:drawable="@drawable/ic_launcher_foreground" />
-</adaptive-icon>
\ No newline at end of file
+</adaptive-icon>
```
Comment (at-vinhhuynh) [inline app/src/main/res/mipmap-anydpi-v26/ic_launcher_round.xml]: revert this file

File: app/src/main/res/mipmap-anydpi-v26/ic_launcher.xml [hunk 18]
```diff
@@ -2,4 +2,4 @@
 <adaptive-icon xmlns:android="http://schemas.android.com/apk/res/android">
     <background android:drawable="@drawable/ic_launcher_background" />
     <foreground android:drawable="@drawable/ic_launcher_foreground" />
-</adaptive-icon>
\ No newline at end of file
+</adaptive-icon>
```
Comment (at-vinhhuynh) [inline app/src/main/res/mipmap-anydpi-v26/ic_launcher.xml]: revert this file

File: app/src/at-phongle/res/layout/item_post.xml [hunk 19]
```diff
@@ -0,0 +1,66 @@
+<?xml version="1.0" encoding="utf-8"?>
```
Comment (at-vinhhuynh) [inline app/src/at-phongle/res/layout/item_post.xml]: change file name to `item_list_post`

**Comment by at-phongle** [comment]
@at-vinhhuynh i checked comment . please review this . Thanks

File: app/src/at-phongle/java/vn/asiantech/internship/models/Post.java [hunk 20]
```diff
@@ -0,0 +1,39 @@
...
+    }
+
+    public int getLikeNumber() {
+        return likeNumber;
+    }
+
+    public void setLikeNumber(int likeNumber) {
+        this.likeNumber = likeNumber;
+    }
+
```
Comment (at-vinhhuynh) [inline app/src/at-phongle/java/vn/asiantech/internship/models/Post.java]: remove this line

File: app/src/at-phongle/res/layout/activity_post.xml [hunk 21]
```diff
@@ -0,0 +1,6 @@
+<?xml version="1.0" encoding="utf-8"?>
+<android.support.v7.widget.RecyclerView xmlns:android="http://schemas.android.com/apk/res/android"
+    android:id="@+id/recyclerViewPosts"
```
Comment (at-vinhhuynh) [inline app/src/at-phongle/res/layout/activity_post.xml]: change id to `recyclerViewPost`

File: app/src/at-phongle/res/values/colors.xml [hunk 22]
```diff
@@ -3,8 +3,12 @@
     <color name="colorPrimary">#3F51B5</color>
     <color name="colorPrimaryDark">#303F9F</color>
     <color name="colorAccent">#FF4081</color>
+    <color name="greenLike">#32CD32</color>
+    <color name="redDislike">#ce253c</color>
+    <color name="snow">#FAFAFA</color>
+    <color name="black">#000000</color>
+    <color name="bg_recyclerView">#ece8e8</color>
```
Comment (at-vinhhuynh) [inline app/src/at-phongle/res/values/colors.xml]: change resource name to `colorBgRecyclerView`

**Comment by at-phongle** [comment]
@at-vinhhuynh i checked comment . please check it. thanks

**Comment by at-vinhhuynh** [comment]
@at-huongnguyen 
I finish review this PR, please check it again.
thank you!

**Pull request merged**